apps/api/src/app/widgets/usecases/mark-all-message-as-seen/mark-all-message-as-seen.usecase.ts

```diff
@@ -0,0 +1,34 @@
+import { Inject, Injectable } from '@nestjs/common';
+import { MessageRepository } from '@novu/dal';
+import { AnalyticsService } from '../../../shared/services/analytics/analytics.service';
+import { QueueService } from '../../../shared/services/queue';
+import { ANALYTICS_SERVICE } from '../../../shared/shared.module';
+import { MarkAllMessageAsSeenCommand } from './mark-all-message-as-seen.command';
+
+@Injectable()
+export class MarkAllMessageAsSeen {
+  constructor(
+    private messageRepository: MessageRepository,
+    private queueService: QueueService,
+    @Inject(ANALYTICS_SERVICE) private analyticsService: AnalyticsService
+  ) {}
+
+  async execute(command: MarkAllMessageAsSeenCommand): Promise<number> {
+    const response = await this.messageRepository.markAllUnseenAsSeen(command.subscriberId);
```

Comment:
Can we add an `_environomentId` to the query, we work in a multi tenant environment and the subscriberId is not unique here :)

Reply:
Sure I will add